http.js

```diff
@@ -105,42 +105,26 @@ Http.prototype._pushToLog = function (logEntry) {
   this._log.push(logEntry);
 };
 
-Http.prototype._buildParams = function (param1, param2) {
-  var params;
-
-  params = {
-    opts: {},
-    callback: undefined
-  };
-  params.callback = param2;
-
-  if (_.isString(param1)) {
-    params.opts.url = param1;
-  } else {
-    params.opts = param1;
-  }
-
-  return params;
-};
-
-Http.prototype.request = function (method, opts, callback) {
+Http.prototype.request = function(method, opts, callback) {
   var _this, data, url, finalOpts, makeRequest;
 
   _this = this;
   makeRequest = Q.defer();
 
-  if (!opts.url) {
+  url = _.isString(opts) ? opts : opts.url;
+
+  if (_.isUndefined(url)) {
     throw new Error('Url is not set');
   }
 
-  url = opts.url;
+  opts = _.isObject(opts) ? opts : {};
   data = opts.data || null;
 
   finalOpts = _.omit(opts, ['data', 'url']);
   finalOpts.cookies = _.extend(this._cookieJar, finalOpts.cookies);
 
-  needle.request(method, url, data, finalOpts, function (err, res, body) {
-    _this._interceptResponse(err, res, body, opts.url, data, makeRequest, callback);
+  needle.request(method, url, data, finalOpts, function(err, res, body) {
```

Comment:
According to our coding style there should be a space between `function` and `(err, res, body)`